source/extensions/access_loggers/open_telemetry/grpc_access_log_impl.cc

```diff
@@ -44,6 +35,34 @@ opentelemetry::proto::common::v1::KeyValue getStringKeyValue(const std::string&
 
 } // namespace
 
+GrpcAccessLoggerImpl::GrpcAccessLoggerImpl(
+    const Grpc::RawAsyncClientSharedPtr& client,
+    const envoy::extensions::access_loggers::open_telemetry::v3::OpenTelemetryAccessLogConfig&
+        config,
+    Event::Dispatcher& dispatcher, const LocalInfo::LocalInfo& local_info, Stats::Scope& scope)
+    : GrpcAccessLogger(
+          config.common_config(), dispatcher, scope, std::nullopt,
+          std::make_unique<Common::UnaryGrpcAccessLogClient<ExportLogsServiceRequest,
+                                                            ExportLogsServiceResponse>>(
+              client,
+              *Protobuf::DescriptorPool::generated_pool()->FindMethodByName(
+                  "opentelemetry.proto.collector.logs.v1.LogsService.Export"),
+              GrpcCommon::optionalRetryPolicy(config.common_config()), genOTelCallbacksFactory())),
+      stats_({ALL_GRPC_ACCESS_LOGGER_STATS(POOL_COUNTER_PREFIX(scope, GRPC_LOG_STATS_PREFIX))}) {
+  initMessageRoot(config, local_info);
+}
+
+std::function<GrpcAccessLoggerImpl::OTelLogRequestCallbacks*()>
+GrpcAccessLoggerImpl::genOTelCallbacksFactory() {
+  return [this]() {
+    OTelLogRequestCallbacks* ptr = new OTelLogRequestCallbacks(
+        this->stats_, this->batched_log_entries_,
+        [this](OTelLogRequestCallbacks* p) { this->callbacks_.erase(p); });
```

Comment:
I do not think you can do it this way. You need to first find `p` in callbacks_ and then delete at the iterator if it was found.

Reply:
Are you asking about:
1) this API doesn't work?
2) or we need to do a sanity check on the existence?

For 1), it should work. absl::flat_hash_map has the api to directly remove the key
For 2), I just added the sanity check in case it is called twice.